tests/filecheck/backend/riscv/optimisation_riscv.mlir

```diff
@@ -1,33 +1,11 @@
 // RUN: xdsl-opt -p optimise-riscv %s | filecheck %s
 
 builtin.module {
-  "riscv_func.func"() ({
-  ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
-    %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
-    %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
-    %4 = riscv.li 1 : () -> !riscv.reg<t0>
-    %5 = riscv.li 1 : () -> !riscv.reg<t1>
-    %6 = "riscv_scf.for"(%2, %3, %4, %5) ({
-    ^1(%7 : !riscv.reg<a0>, %8 : !riscv.reg<t1>):
-      %9 = riscv.add %8, %7 : (!riscv.reg<t1>, !riscv.reg<a0>) -> !riscv.reg<t1>
-      "riscv_scf.yield"(%9) : (!riscv.reg<t1>) -> ()
-    }) : (!riscv.reg<a0>, !riscv.reg<a1>, !riscv.reg<t0>, !riscv.reg<t1>) -> !riscv.reg<t1>
-    %10 = riscv.mv %6 : (!riscv.reg<t1>) -> !riscv.reg<a0>
-    "riscv_func.return"(%10) : (!riscv.reg<a0>) -> ()
-  }) {"sym_name" = "funky"} : () -> ()
+  %0, %1 = "test.op"() : () -> (!riscv.reg<a0>, !riscv.reg<a1>)
+  %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+  %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
```

Comment:
This would test for the negative case, if we move to a different register, we don't eliminate the mv
```suggestion
  %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a2>
```

Reply:
Negative test cases have been added.